processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
@jihoonson, the fields on these emitted metrics are generally going to be used as dimensions for slice and dice kind of analysis. Imagine them as dimensions in a Druid dataSource, since often, that's where they are going (Druid itself is a common sink for Druid metrics). So the question is "what are cluster operators going to want to filter/group on when analyzing Druid query metrics".

The existing Druid query metrics schemas were designed for single-dataSource queries, but even so, it's probably best to extend them "lightly" so metrics from join queries mix well with metrics from other kinds of queries. (the user might want to know "average query time for a certain dataSource" and that should include join queries).

Keep in mind we don't need to emit _everything_ about a query. That's what the request log is for. Query metrics are just for emitting attributes that operators may want to slice/dice on to understand and optimize performance.

I'm not sure about the best design in full, but I think at least, there should be a `dataSource` field (since other query types have it) that maybe either includes the name of the "main" distributed dataSource, or maybe includes a comma separated list of all dataSources involved in the query (like `DataSourceUtil.getMetricName` would do).

For `interval` / `duration` I don't have a very strong opinion. I share @leventov's view that as-is, interval is not awfully useful for existing query types, and `duration` is somewhat more useful. Maybe for join queries, `duration` should be the "umbrella duration" (coalesce the intervals across all dataSources, then add up the durations) and we could add a new multi-value field `dataSourcesAndDurations` that has pairs like `dataSourceOne,PT4H`?